bundles/framework/layerselector2/Flyout.js

```diff
@@ -240,11 +241,13 @@ Oskari.clazz.define('Oskari.mapframework.bundle.layerselector2.Flyout',
         /**
          * Hande selected filter request
          * @method  public enableFilter
-         * @param  {String} selectedFilter selected filter, can be a 'stats', 'newest' or 'publishable'
+         * @param  {String} selectedFilter selected filter, can be a 'vectors', 'newest' or 'publishable'
          */
         enableFilter: function(selectedFilter) {
+            // Trigger first founded tab filter --> this activate/deactivate filterings also another tabs
             var me = this,
-                filterButton = jQuery('.layer-filter .filter-'+selectedFilter).first(),
+                filterContainer = (me.layerTabs.length>0) ? me.layerTabs[0].getTabPanel().getContainer().find('.layerselector2-layer-filter') : jQuery('<div></div>'),
```

Comment:
Why is this always done on the first of the layerTabs? Looks like it could be a bug. Should it be the selected tab?

Reply:
Also why would you wan't to create a new div if there are no layerTabs? Just stop the execution if there are none?